src/App.tsx

```diff
@@ -1,26 +1,123 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { Errors } from './types/ErrorType';
+import { TodoItem } from './components/TodoItem';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoHeader } from './components/TodoHeader';
+import { ErrorNotification } from './components/ErrorNotification';
+import { getTodos, addTodo, USER_ID, deleteTodo } from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<FilterType>(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Default);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState<number[]>([]);
+
+  const filteredTodos = todos?.filter(todo => {
+    switch (status) {
+      case FilterType.Completed:
+        return todo.completed;
+      case FilterType.Active:
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const unCompletedTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const completedTodos = useMemo(
+    () => todos.filter(todo => todo.completed).length,
+    [todos],
+  );
+
+  const onAddTodo = async (todoTitle: string) => {
+    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (err) {
+      setErrorMessage(Errors.AddTodo);
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const onRemoveTodo = async (todoId: number) => {
+    setLoadingTodo(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (err) {
+      setErrorMessage(Errors.DeleteTodo);
+      throw err;
```

Comment:
You should not throw error here, because you don't use `try/catch` in place where you use this function